scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -1751,3 +1752,7 @@ def custom_strategy_fn(candidate, population, rng=None):
                 bounds,
                 strategy=custom_strategy_fn
             )
+
+    def test_reference_cycles(self):
+        with assert_deallocated(DifferentialEvolutionSolver, rosen, [(0, 10)]*2):
+            pass
```

Comment:
ah, looks like you pushed this in just as I was about to merge.

Anyway, confirming this fails before and passes after the source patch is applied for me locally on ARM Mac, so even better.

I suspect we *should* now let the CI flush through again though, since reference cycling testing can be tricky?